waku/v2/protocol/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -364,6 +366,15 @@ method init*(g: OnchainGroupManager): Future[void] {.async.} =
   let chainId = await ethRpc.provider.eth_chainId()
   g.chainId = some(chainId)
 
+  if g.ethPrivateKey.isSome():
+    let pk = g.ethPrivateKey.get()
+    let pkParseRes = keys.PrivateKey.fromHex(pk)
+    if pkParseRes.isErr():
+      raise newException(ValueError, "could not parse the private key")
+    ethRpc.privateKey = some(pkParseRes.get())
+    ethRpc.defaultAccount = ethRpc.privateKey.get().toPublicKey().toCanonicalAddress().Address
```

Comment:
weird that we need to set the defaultAccount here, without which transactions use the zero address as the from address instead of deriving it from the pk.